packages/lean-imt/src/lean-imt.ts

```diff
@@ -233,6 +233,52 @@ export default class LeanIMT<N = bigint> {
         this._nodes[this.depth] = [node]
     }
 
+    /**
+     * Updates N leaves all at once.
+     * It is more efficient than using the {@link LeanIMT#update} method N times because it
+     * prevents updating middle nodes several times. This would happen when updating leaves
+     * with common ancestors. However, it doesn't offer a better worst-case time complexity.
+     * @param indices The list of indices of the respective leaves.
+     * @param leaves The list of leaves to be updated.
+     */
+    public updateMany(indices: number[], leaves: N[]) {
+        requireDefined(leaves, "leaves")
+        requireDefined(indices, "indices")
+        requireArray(leaves, "leaves")
+        requireArray(indices, "indices")
+
+        if (leaves.length !== indices.length) {
```

Comment:
It may be necessary to check if there're duplicate indices. Is there a test to see what happens?

Reply:
In principle, I didn't think passing repeated indices should be a problem. What it does with the current implementation is only apply the last change of each index given the order of the arrays passed as parameters. That is, if you pass ([1, 1], [a, b]), it will update the index 1 to b, and if you pass ([1, 1], [b, a]) it will update the index 1 to a. I just added a test that verifies this behavior.

If desired, I could add a check to throw an error when detecting repeated indices.

Reply:
Replied here: https://github.com/privacy-scaling-explorations/zk-kit/pull/314#discussion_r1740607650